testament/testament.nim

```diff
@@ -637,156 +658,195 @@ proc loadSkipFrom(name: string): seq[string] =
       result.add sline
 
 proc main() =
-  azure.init()
-  backend.open()
-  var optPrintResults = false
-  var optFailing = false
-  var targetsStr = ""
-  var isMainProcess = true
-  var skipFrom = ""
-
-  var p = initOptParser()
-  p.next()
-  while p.kind in {cmdLongOption, cmdShortOption}:
-    case p.key.normalize
-    of "print": optPrintResults = true
-    of "verbose": optVerbose = true
-    of "failing": optFailing = true
-    of "pedantic": discard # deadcode refs https://github.com/nim-lang/Nim/issues/16731
-    of "targets":
-      targetsStr = p.val
-      gTargets = parseTargets(targetsStr)
-      targetsSet = true
-    of "nim":
-      compilerPrefix = addFileExt(p.val.absolutePath, ExeExt)
-    of "directory":
-      setCurrentDir(p.val)
-    of "colors":
-      case p.val:
-      of "on":
-        useColors = true
-      of "off":
-        useColors = false
-      else:
-        quit Usage
-    of "batch":
-      testamentData0.batchArg = p.val
-      if p.val != "_" and p.val.len > 0 and p.val[0] in {'0'..'9'}:
-        let s = p.val.split("_")
-        doAssert s.len == 2, $(p.val, s)
-        testamentData0.testamentBatch = s[0].parseInt
-        testamentData0.testamentNumBatch = s[1].parseInt
-        doAssert testamentData0.testamentNumBatch > 0
-        doAssert testamentData0.testamentBatch >= 0 and testamentData0.testamentBatch < testamentData0.testamentNumBatch
-    of "simulate":
-      simulate = true
-    of "megatest":
-      case p.val:
-      of "on":
-        useMegatest = true
-      of "off":
-        useMegatest = false
-      else:
-        quit Usage
-    of "backendlogging":
-      case p.val:
-      of "on":
-        backendLogging = true
-      of "off":
-        backendLogging = false
+  ## Define CLI Options/Args Parsing loops
+  template parseOptions: untyped {.dirty.} =
```

Comment:
Not sure about templates here, wouldn't named blocks do the trick?

Reply:
I prefer being able to fold what should have probably been independent code blocks/procedures from the beginning at the head of the procedure since the case action is imo the most significant code block.

Reply:
Then it should take a parser argument.